blocks/columns/columns.js

```diff
@@ -1,18 +1,32 @@
 export default function decorate(block) {
   const cols = [...block.firstElementChild.children];
   block.classList.add(`columns-${cols.length}-cols`);
+  let downloadableColumns = false;
 
   // setup image columns
   [...block.children].forEach((row) => {
     [...row.children].forEach((col) => {
       const pic = col.querySelector('picture');
+      const link = col.querySelector('a');
+
+      if (pic && link && link.href
+        && new URL(link.href).pathname === new URL(link.innerText).pathname) {
+        link.innerHTML = '';
+        link.appendChild(pic.cloneNode(true));
```

Comment:
nit: Couldn't you just append the `pic` element itself and move it this way, instead of cloning and deleting? Or am I missing something here?

Reply:
Oh yes; thank you :)